app/models/transformation_mapping_item.rb

```diff
@@ -24,4 +31,36 @@ def destination_cluster
       errors.add(:destination, "EMS type of destination cluster or cloud tenant must be in: #{destination_types}")
     end
   end
+
+  def source_datastore
+    source_storage = source
+
+    cluster_storages = source.hosts # Get hosts using this source storage
+                             .collect(&:ems_cluster) # How many clusters does each host have
+                             .collect(&:storages).flatten # How many storages each host is mapped to that belong to the cluster
```

Comment:
I think you want to go through the `transformation_mapping.transformation_mapping_items` looking for where the source type is EmsCluster, then check that this storage is in _that_ cluster, not that the storage is in its own cluster

Reply:
I am going to seek @djberg96 help for how to achieve this.   The transformation_mapping table has no column/reference for transforamtion_mapping_items.   May be in the Ruby class, I can get clusters using TransformationMappingItem.getParent.getTransformationMappingItems.where(:source_type => "EmsCluster").  Will update what if find.

Reply:
@thearifismail but the transformation_mapping_items table has a foreign key to the transformation_mapping, this is how the `belongs_to :transformation_mapping` works.  Try creating one of these and then play around a bit in a rails console to see what is available, but from the here you should be able to do: `transformation_mapping.transformation_mapping_items.where(:source_type => "EmsCluster")`